libcodechecker/cmd/cmd_line_client.py

```diff
@@ -686,72 +853,77 @@ def get_run_tag(client, run_ids, tag_name):
-        cmp_data.diffType = ttypes.DiffType.UNRESOLVED
-    elif 'resolved' in args:
-        cmp_data.diffType = ttypes.DiffType.RESOLVED
-
-    results = []
-    if os.path.isdir(args.newname):
-        # If newname is a valid directory we assume that it is a report dir and
-        # we are in local compare mode.
-        results = get_diff_report_dir(client, base_ids,
-                                      os.path.abspath(args.newname),
-                                      cmp_data)
-    else:
-        run_with_tag = args.newname.split(':')
-        new_run_name = run_with_tag[0]
+        # Set base run tag if it is available.
         run_tag_name = run_with_tag[1] if len(run_with_tag) > 1 else None
+        run_tags = None
+        if run_tag_name:
+            tag = get_run_tag(client, run_ids, run_tag_name)
+            run_tags = [tag.id] if tag else None
 
-        new_runs = get_runs(client, [new_run_name])
-        new_ids = map(lambda run: run.runId, new_runs)
-
-        if len(new_ids) == 0:
+        if not run_ids:
             LOG.warning(
-                "No run names match the given pattern: " + args.newname)
+                "No run names match the given pattern: " + run_arg_with_tag)
             sys.exit(1)
 
-        LOG.info("Matching new runs: " +
-                 ', '.join(map(lambda run: run.name, new_runs)))
-
-        cmp_data.runIds = new_ids
-        if run_tag_name:
-            tag = get_run_tag(client, new_ids, run_tag_name)
-            cmp_data.runTag = [tag.id] if tag else None
+        LOG.info("Matching runs: %s",
+                 ', '.join(map(lambda run: run.name, runs)))
 
-        results = get_diff_results(client, base_ids, report_filter, cmp_data)
+        return run_ids, run_tags
 
-    if len(results) == 0:
-        LOG.info("No results.")
+    def print_diff_results(reports):
+        """
+        Print the results.
+        """
+        if reports:
+            print_reports(client, reports, args.output_format)
+        else:
+            LOG.info("No results.")
+
+    client = None
+
+    # We set up the client if we are not comparing two local report directory.
+    if not os.path.isdir(args.basename) or not os.path.isdir(args.newname):
+        client = setup_client(args.product_url)
+
+    if os.path.isdir(args.basename) and os.path.isdir(args.newname):
+        reports = get_diff_local_dirs(args.basename, args.newname)
+        print_diff_results(reports)
+        LOG.info("Compared two local report directories %s and %s",
+                 os.path.abspath(args.basename),
+                 os.path.abspath(args.newname))
+    elif os.path.isdir(args.newname):
+        reports = get_diff_remote_run_local_dir(client,
+                                                args.basename,
+                                                os.path.abspath(args.newname))
+        print_diff_results(reports)
+        LOG.info("Compared remote run %s and local report directory %s",
```

Comment:
remote run can be multiple runs if a user gave a wildcard pattern like `run*`, the printed text should reflect that.
The same issue applies to other places where run names are used for comparison.

It could help if we would print out the matching run names in these cases.